bumble/profiles/hap.py

```diff
@@ -447,16 +448,16 @@ def get_op_index(op: PresetChangedOperation) -> int:
             return op.additional_parameters
 
         op_list.sort(key=get_op_index)
-        # If the ATT bearer is terminated before all notifications or indications are sent, then the server shall consider the Preset Changed operation aborted and shall continue the operation when the client reconnects.
-        while len(op_list) > 0:
+        # If the ATT bearer is terminated before all notifications or indications are
+        # sent, then the server shall consider the Preset Changed operation aborted and
+        # shall continue the operation when the client reconnects.
+        for i, op in enumerate(op_list):
             try:
                 await connection.device.indicate_subscriber(
                     connection,
                     self.hearing_aid_preset_control_point,
-                    value=op_list[0].to_bytes(len(op_list) == 1),
+                    value=op.to_bytes(i == len(op_list) - 1),
                 )
-                # Remove item once sent, and keep the non sent item in the list
-                op_list.pop(0)
```

Comment:
Why don't we need to remove it from the list anymore ?

My understanding is that this list is sent anytime the device reconnect, and if we keep it in the list, connecting twice will send the list of event twice too, unless we remove the event when the indication got acknowledge (hence why we don't clear all at once, but only one at a time)